[DOCS] Reviews ML decider conceptual docs (#69524)
elastic/elasticsearch: docs/reference/autoscaling/deciders/machine-learning-decider.asciidoc

[[autoscaling-machine-learning-decider]]
=== Machine learning decider

The {ml} decider (`ml`) calculates the memory required to run {ml} jobs.

The {ml} decider is enabled for policies governing `ml` nodes.

NOTE: For {ml} jobs to open when the cluster is not appropriately
scaled, set `xpack.ml.max_lazy_ml_nodes` to the largest number of possible {ml} 

> jobs (refer to <<advanced-ml-settings>> for more info). In {ess} this is 

Reviewer: Not mandatory,  but I prefer the full word:
```suggestion
jobs (refer to <<advanced-ml-settings>> for more information). In {ess} this is 
```

automatically set.

[[autoscaling-machine-learning-decider-settings]]
==== Configuration settings

Both `num_anomaly_jobs_in_queue` and `num_analytics_jobs_in_queue` are designed 

> to delay a scale-up event. They indicate how many jobs of that type can be 

Reviewer: Not mandatory, but I like putting the condition at the beginning of the sentence, for example:
```suggestion
to delay a scale-up event. If the cluster is too small, these settings indicate how many jobs of each type can be 
```

> unassigned from a node due to the cluster being too small. Both settings are 

Reviewer: If you accept the previous suggestion, here's the end of the sentence:
```suggestion
unassigned from a node. Both settings are 
```

only considered for jobs that can be opened given the current scale. If a job is 
too large for any node size or if a job can't be assigned without user 
intervention (for example, a user calling `_stop` against a real-time 
{anomaly-job}), the numbers are ignored for that particular job.

`num_anomaly_jobs_in_queue`::
(Optional, integer)
Specifies the number of queued {anomaly-jobs} to allow. Defaults to `0`.

`num_analytics_jobs_in_queue`::
(Optional, integer)
Specifies the number of queued {dfanalytics-jobs} to allow. Defaults to `0`.

`down_scale_delay`::
(Optional, <<time-units,time value>>)
Specifies the time to delay before scaling down. Defaults to 1 hour. If a scale 

> down is possible for the entire time window, then a scale down is requested. If 

Reviewer: > down is possible for the entire time window...

This is probably a question for @benwtrent, but something about this description is unclear IMO. Is it saying that it somehow knows in advance that a scale up is required within the specified time interval and if so, won't scale down?

Reviewer 2: > Is it saying that it somehow knows in advance that a scale up is required within the specified time interval and if so, won't scale down?

No, the deciders are periodically polled for scaling decisions. If during all these polls (over the span of 1 hour), a scale down is possible, then a scale down will be requested.

Author: If I understand correctly, then it says that if a scale-up is not required during the time of `down_scale_delay`, then it will scale down after the time is up. If we use the default value for example, then if scale-up is not needed during one hour, then it scales down after one hour. If scale-up is needed, then the timer starts again.

Reviewer 2: yes
